src/ripple/app/data/SociDB.cpp

```diff
@@ -84,55 +84,63 @@ std::string SociConfig::connectionString () const
     return connectionString_;
 }
 
-void SociConfig::open(soci::session& s) const
+void SociConfig::open (soci::session& s) const
 {
     s.open (backendFactory_, connectionString ());
 }
 
-void open(soci::session& s,
-          BasicConfig const& config,
-          std::string const& dbName)
+void open  (soci::session& s,
+            BasicConfig const& config,
+            std::string const& dbName)
 {
-    SociConfig c(config, dbName);
-    c.open(s);
+    SociConfig (config, dbName).open(s);
 }
 
-void open(soci::session& s,
-          std::string const& beName,
-          std::string const& connectionString)
+void open (soci::session& s,
+           std::string const& beName,
+           std::string const& connectionString)
 {
     if (beName == "sqlite")
-    {
         s.open(soci::sqlite3, connectionString);
-        return;
-    }
     else
-    {
         throw std::runtime_error ("Unsupported soci backend: " + beName);
-    }
+}
+
+static
+sqlite_api::sqlite3* getConnection (soci::session& s)
+{
```

Comment:
@vinniefalco Thanks for the guidance. I'm making a note that new elements in cpp files should prefer to use the current top-level namespace (`ripple`) in cpp files and not use `detail` namespaces or anon namespaces.